src/js/controllers/messageController.js

```diff
@@ -99,30 +101,50 @@ export const messageScreen = (state) => {
     document.querySelector('.type-field').addEventListener('click', e => {
         e.preventDefault();
         const btn = e.target.closest('#the-btn');
-        console.log(btn);
 
         if (btn){
-            const now = new firebase.firestore.Timestamp.now().toDate();
-            const message = btn.parentNode.parentNode.childNodes[3].childNodes[1].value;
-            //console.log(message);
-            //console.log(state.messages.selfPos);
-            state.messages.sendMessage(message, state.messages.currentChatId,now);
-        }
-        
+            const typeBox = btn.parentNode.parentNode.childNodes[3].childNodes[1];
+            if (typeBox.value){
+                const now = new firebase.firestore.Timestamp.now().toDate();
+                const message = typeBox.value;
+                state.messages.sendMessage(message, state.messages.currentChatId,now);
+                typeBox.value = '';
+                document.body.scrollTop = document.body.scrollHeight;
+            }
+        }        
+
     });
 
     const searchResUI = document.querySelector('.search-results');
 
     document.querySelector('.search-btn').addEventListener('click', async e =>  {
         e.preventDefault();
         const btn = e.target.closest('.search-btn');
-        console.log(btn);
         if(btn){
             const searchContent = btn.parentNode.childNodes[3].value;
             await state.messages.getSearchResults(searchContent);
+            contactsView.renderSearchResults(state.messages.searchRes);
             searchResUI.classList.remove("invisible");
         }
 
     })
 
-};
+    searchResUI.addEventListener('click', async e=>{
+        const btn = e.target.closest('.search-result-person');
+        const chatter = btn.parentNode.parentNode.childNodes[1].childNodes[3];
+        console.log(chatter);
+
+        if (btn){
+            state.messages.chatExists = false;
+            await state.messages.doesChatExists(btn.id,state.user.uid);
+            if (state.messages.chatExists){
```

Comment:
rather than making the call and then retrieving the value from `state`, it'd be better to just return the result so that you can do something like `if (await state.messages.doesChatExist(...)) {...}`

Reply:
We had most of our mode code structured in this way. Call the async function first, then use the state attributes going forward. I think, to keep this consistency, we may not want to change it.

Reply:
Sure, sounds good